src/app/features/user-profile/user-profile.service.ts

```diff
@@ -0,0 +1,446 @@
+ * Handles profile CRUD operations and switching
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class UserProfileService {
+  private readonly _storageService = inject(UserProfileStorageService);
+  private readonly _pfapiService = inject(PfapiService);
+  private readonly _snackService = inject(SnackService);
+  private readonly _injector = inject(Injector);
+  private readonly _globalConfigService = inject(GlobalConfigService);
+
+  // Current profile metadata
+  private readonly _metadata = signal<ProfileMetadata | null>(null);
+  readonly metadata = this._metadata.asReadonly();
+
+  // Currently active profile
+  readonly activeProfile = signal<UserProfile | null>(null);
+
+  // All available profiles
+  readonly profiles = signal<UserProfile[]>([]);
+
+  /**
+   * Initialize the profile system
+   * Called during app startup
+   */
+  async initialize(): Promise<void> {
+    Log.log('UserProfileService: Initializing profile system');
+
+    try {
+      // Load existing profile metadata
+      let metadata = await this._storageService.loadProfileMetadata();
+
+      // If no metadata exists, check if there's existing user data
+      if (!metadata) {
+        Log.log('UserProfileService: No profile metadata found, migrating existing data');
+        metadata = await this._migrateExistingDataToDefaultProfile();
+      }
+      // Profile data is only loaded when switching between profiles.
+      // On normal startup, the app loads from its regular database.
+      // The profile storage is just a backup copy for switching, leveraging existing import/export functions.
+
+      // Set metadata and update signals
+      this._metadata.set(metadata);
+      this.profiles.set(metadata.profiles);
+
+      // Set active profile
+      const activeProfile = metadata.profiles.find(
+        (p) => p.id === metadata.activeProfileId,
+      );
+      if (activeProfile) {
+        this.activeProfile.set(activeProfile);
+        Log.log(
+          `UserProfileService: Active profile set to "${activeProfile.name}" (${activeProfile.id})`,
+        );
+      } else {
+        Log.warn(
+          'UserProfileService: Active profile not found in metadata, using first profile',
+        );
+        this.activeProfile.set(metadata.profiles[0]);
+      }
+    } catch (error) {
+      Log.err('UserProfileService: Failed to initialize', error);
+      // create a default profile and continue
+      const defaultMetadata = this._storageService.createDefaultProfileMetadata();
+      this._metadata.set(defaultMetadata);
+      this.profiles.set(defaultMetadata.profiles);
+      this.activeProfile.set(defaultMetadata.profiles[0]);
+    }
+  }
+
+  /**
+   * Create a new profile
+   */
+  async createProfile(name: string): Promise<UserProfile> {
+    const metadata = this._metadata();
+    if (!metadata) {
+      throw new Error('Profile system not initialized');
+    }
+
+    if (!name || name.trim().length === 0) {
```

Comment:
The first part of the condition is unnecessary; I would suggest removing it
```suggestion
    if (name.trim().length === 0) {
```